src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CaseDocumentsCIC.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.hmcts.sptribs.ciccase.model;
+
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import uk.gov.hmcts.ccd.sdk.api.CCD;
+import uk.gov.hmcts.ccd.sdk.type.ListValue;
+import uk.gov.hmcts.sptribs.ciccase.model.access.DefaultAccess;
+import uk.gov.hmcts.sptribs.document.model.CICDocument;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Objects.isNull;
+import static java.util.UUID.randomUUID;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toCollection;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.springframework.util.CollectionUtils.isEmpty;
+import static uk.gov.hmcts.ccd.sdk.type.FieldType.Collection;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class CaseDocumentsCIC {
+
+    @CCD(
+        label = "Add a file" + "\nUpload a file to the system",
+        typeOverride = Collection,
+        typeParameterOverride = "CICDocument",
+        access = {DefaultAccess.class}
+    )
+    private List<ListValue<CICDocument>> applicantDocumentsUploaded;
+
+
+    public static <T> List<ListValue<T>> addDocumentToTop(final List<ListValue<T>> documents, final T value) {
+        return addDocumentToTop(documents, value, null);
+    }
+
+    public static <T> List<ListValue<T>> addDocumentToTop(final List<ListValue<T>> documents, final T value, final String id) {
+        final var listItemId = isBlank(id) ? String.valueOf(randomUUID()) : id;
+        final var listValue = new ListValue<T>(listItemId, value);
+        final List<ListValue<T>> list = isEmpty(documents) ? new ArrayList<>() : new ArrayList<>(documents);
```

Comment:
Very minor quibble, but maybe call what's in the list ListItem rather than ListValue?